### PR TITLE
Loader doc

### PR DESCRIPTION
**Pull request by cachedout**
Some docs for the loader and a note in the public documtation on what `opts` is and how to create one from scratch.

### CODE REVIEW

**Comment by cachedout** [comment]
Test failures unrelated.

File: doc/ref/internals/opts.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.. _opts:
+
+====================
+Salt opts dictionary
+====================
+
+It is very common in the Salt codebase to see `opts` referred to in a number of
+contexts.
+
+For example, it can be seen as `__opts__` in certain cases, or simple as `opts`
```
Comment (rallytime) [inline doc/ref/internals/opts.rst]: simply?

Comment (rallytime) [inline doc/ref/internals/opts.rst]: Nevermind. I just grabbed it. Thanks!

**Comment by cachedout** [comment]
Thanks. Due to jet lag, I'm only reading every other character. ;]

**Pull request merged**